### PR TITLE
Preregister approvals via email and register lifebank

### PR DESCRIPTION
**Pull request by AngeloCG97**
## Preregister approvals via email

#345 

### What does this PR do?
- Coding desing html email form
- Send Lifebank registration data via email.
- Create page to start process register lifebank
- Register LB in users table
- Create LB account on blockchain

### How should this be tested?
- Run make run
- go to localhost:3000
- register lifebank

### CODE REVIEW

File: hapi/src/api/account.api.js [hunk 1]
```diff
@@ -272,6 +304,30 @@ const grantConsent = async account => {
   return consentTransaction
 }
 
+const formatSchedule = (schedule) => {
+  let scheduleFormat = ''
+
+  var hours
```
Comment (JustinCast) [inline hapi/src/api/account.api.js:310]: User let or const instead of var

File: hapi/src/api/account.api.js [hunk 2]
```diff
@@ -272,6 +304,30 @@ const grantConsent = async account => {
...
+  for (hours of schedule) {
```
Comment (JustinCast) [inline hapi/src/api/account.api.js:311]: Unnecessary curly braces here

File: webapp/src/routes/Signup/Signup.js [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+      if (data) {
+        data.preregister_lifebank.length === 0 && data.user.length === 0
+          ? setEmailValid(true)
+          : setEmailValid(false)
+
+        setcheckEmailLoaded(true)
+      }
+    }
+
+    if (regularExpresion.test(user?.email)) {
```
Comment (JustinCast) [inline webapp/src/routes/Signup/Signup.js:266]: Unnecessary curly braces here

File: webapp/src/routes/Signup/Signup.js [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+          </Alert>
+        )}
+      </>
+    )
+  }
+
+  return (
+    <Grid container className={classes.gridContainer}>
+      <Grid item xs={12} sm={8} md={6} className={classes.register}>
+        <Typography variant="h1">{`${t('miscellaneous.step')} ${activeStep + 1
```
Comment (JustinCast) [inline webapp/src/routes/Signup/Signup.js:339]: Why the line breaks?

**Review by JustinCast — CHANGES_REQUESTED** [review]

File: hapi/src/api/account.api.js [hunk 5]
```diff
@@ -57,7 +58,38 @@ const create = async ({ role, email, name, secret }) => {
 
   await historyApi.insert(transaction)
 
-  //mailApi.sendVerificationCode(email, verification_code)
+  mailApi.sendVerificationCode(email, verification_code)
```
Comment (JustinCast) [inline hapi/src/api/account.api.js:61]: @AngeloCG97 what if we wrap this call on a try-catch statement?

**Review by JustinCast — CHANGES_REQUESTED** [review]

File: hapi/src/api/account.api.js (hunk 5 above)
Comment (AngeloCG97) [reply hapi/src/api/account.api.js:61]: ok I'll do it now

File: hapi/src/api/account.api.js [hunk 6]
```diff
@@ -272,6 +304,29 @@ const grantConsent = async account => {
...
+  for (hours of schedule)
+    scheduleFormat += `, ${hours.day} ${hours.open} - ${hours.close}`
+
+  return scheduleFormat.replace(',', ' ')
+}
+
+const formatLifebankData = (lifebankData) => {
+  lifebankData.schedule = formatSchedule(JSON.parse(lifebankData.schedule))
+  lifebankData.coordinates = JSON.parse(lifebankData.coordinates)
+  if (lifebankData.immunity_test) lifebankData.immunity_test = 'Yes'
```
Comment (JustinCast) [inline hapi/src/api/account.api.js:320]: What if we move this to lifebank.api.js file?

**Review by JustinCast — CHANGES_REQUESTED** [review]

File: hapi/src/api/account.api.js (hunk 6 above)
Comment (AngeloCG97) [reply hapi/src/api/account.api.js:320]: ok good

**Review by JeanVegaD — APPROVED** [review]

File: hapi/src/api/lifebank.api.js [hunk 7]
```diff
@@ -112,8 +112,32 @@ const signup = async (account, profile) => {
...
+
+  return scheduleFormat.replace(',', ' ')
+}
+
+const formatLifebankData = (lifebankData) => {
+  lifebankData.schedule = formatSchedule(JSON.parse(lifebankData.schedule))
+  lifebankData.coordinates = JSON.parse(lifebankData.coordinates)
+  if (lifebankData.immunity_test) lifebankData.immunity_test = 'Yes'
+  else lifebankData.immunity_test = 'No'
+  if (lifebankData.urgency_level === 1) lifebankData.urgency_level = 'Low'
```
Comment (JustinCast) [inline hapi/src/api/lifebank.api.js:130]: Add blank here. To make it clear

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Pull request merged**